login.js

```diff
@@ -1,23 +1,31 @@
+let isAMatch;
+
 function verifiyPassword(hash) {
-  checkpw("12345678", hash, displayResult, null);
-  return;
+  const passwords = ['greenbear757', 'organicswan134', 'bluemouse519', 'heavytiger886', 'heavyladybug575'];
+
+  isAMatch = [];
+
+  passwords.forEach(
+    (item) => checkpw(item, hash,
+    (result) => {return isAMatch.push(result);}, null));
 }
 
-function displayResult(result) {
+function displayResult() {
   const output = document.getElementById('validation');
-  
-  return result ? output.textContent = 'Welcome Back!' :
-  output.textContent = 'Invalid password';
+
+  return isAMatch.some(
+           (match) => match === true) ?
+            output.textContent = 'Welcome Back!' :
+            output.textContent = 'Invalid password';
```

Comment:
Good, `some()` is the right iterator :+1: 

Regarding the refactoring, it's almost there, I would just keep the function in one line, like this:
```js
  return isAMatch.some((match) => match) ?
    output.textContent = 'Welcome Back!' :
    output.textContent = 'Invalid password';
```

Also remember that because `isAMatch` is an array of booleans, you can just return the item, you don't need to compare it with `true` :+1: